calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/data_collector.py

```diff
@@ -530,3 +585,62 @@ def process_detection(
             return CollectionStatus.ACCEPTED
 
         return CollectionStatus.REDUNDANT
+
+    def process_detection_eval_mode(
+        self,
+        image: np.array,
+        detection: BoardDetection,
+        camera_model: Optional[CameraModel] = None,
+        mode: OperationMode = OperationMode.CALIBRATION,
+    ) -> CollectionStatus:
+        """Evaluate detections mad ein evaluation mode."""
+        # process witout filtering detections
+        self.update_linearity_heatmap(self.linearity_heatmap, detection)
+
+    def get_skew_coverage(self):
+        """Get skew percentage covered from a defined range for the indicators."""
+        # Define the number of intervals
+        num_intervals = 50
+        interval_size = 1 / num_intervals
+        # Create a unique set to store covered intervals
+        covered_intervals = set()
+        # range in radians ToDo: define the range
+        skew_range = np.array([0, 1.04])
+
+        for detection in self.training_data.get_detections():
+            if skew_range[0] <= detection.get_normalized_skew() < skew_range[1]:
+                interval_index = int(detection.get_normalized_skew() / interval_size)
+                covered_intervals.add(interval_index)
+
+        # Calculate the percentage of covered intervals
+        percentage_coverage = len(covered_intervals) / num_intervals  # * 100
+
+        return percentage_coverage
+
+    def get_skew_percentage(self):
+        """Get skew precentage with more useful name ToDo Refactor."""
```

Comment:
TODO(yourname): comment

Reply:
Whenever you address a comment, please reference the commit here. These commits should be as small as possible, so the review process go smoother

Reply:
f2ad5cce4cffc72284d0b575020c40851475872e